Libraries/StyleSheet/processTextShadow.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow strict-local
+ * @format
+ */
+
+import type {ColorValue} from '../StyleSheet/StyleSheet';
+
+import {getColorRegExStrings, namedColors} from '@react-native/normalize-color';
+
+function processTextShadow(textShadow: string): {
+  xOffset?: number,
+  yOffset?: number,
+  blurRadius?: number,
+  color?: ColorValue,
+} {
+  const offsetRegex = '(?:[-+]?\\d+(?:\\.\\d+)?[a-z]*)';
+  const valueSeparator = '\\s+';
+  const twoOrThreeOffsetRegex = `(${offsetRegex}(?:${valueSeparator}${offsetRegex}){1,2})`;
+
+  const colorRegexStrings = getColorRegExStrings();
+  const hex8 = colorRegexStrings.hex8;
+  const hex6 = colorRegexStrings.hex6;
+  const hex4 = colorRegexStrings.hex4;
+  const hex3 = colorRegexStrings.hex3;
+  const hexColor = `(?:(?:${hex8}|${hex6}|${hex3}|${hex4}))\\b`;
+  const colorFunction = `(?:${colorRegexStrings.rgb}|${colorRegexStrings.rgba}|${colorRegexStrings.hsl}|${colorRegexStrings.hsla}|${colorRegexStrings.hwb})`;
+  const namedColor = namedColors.join('|');
```

Comment:
This will be an extremely long regex. Why not check if it matches a word output (`\w+`) and validate afterwards whether it's valid color?

Reply:
Sure, I can change the namedColors Regex to match a word. Also, is validating the colour required? If it is a valid named colour, it will be applied to the shadow. Else, it will be ignored silently.

Reply:
You could still validate that the return of `normalizeKeyword` is not `null` (and can differentiate that from 0x00000000) and throw an error instead of silently ignoring.

Reply:
I kept it like that because the same happens currently when using the `textShadowColor` property of style.